### ISSUE TITLE
[BUG] npm login with scope create .npmrc without @scope:registry=...

### ISSUE THREAD

**Issue by hajekj14** [npm/cli#3292]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior

npm login --registry=http://reg.example.com --scope=@myco

create `~/.npmrc`

```
//reg.example.com/:_authToken="token"
```

### Expected Behavior

npm login --registry=http://reg.example.com --scope=@myco

create `~/.npmrc`

```
@myco:registry=https://reg.example.com/
//reg.example.com/:_authToken="token"
```

### Steps To Reproduce

1. In this environment...
2. With this config...
3. Run '...'
4. See error...

### Environment

- OS: Linux & Win in general
- Node: 
- npm: 7.8.0

**Comment by syakovyn** [comment]
I can confirm the bug still exists on 7.13.0.
Version 7.0.10 is not affected so it looks like a regression.

**Comment by tsuilouis** [comment]
I am using npm v7.20.3 at work and I've encountered this exact same problem. I see that the fix was released in v7.17.0 so I'm wondering if something occurred in between to cause a regression. The problem looks to have even made it all the way to v8.1.4 based on the latest mention.

@hajekj14 @syakovyn were either of you able to verify that the behaviour works as expected after v7.17.0?
